sdk/openai/openai/test/public/utils/asserts.ts

```diff
@@ -0,0 +1,500 @@
+  }
+}
+
+function assertContentFilterResultsForPrompt(cfr: ContentFilterResultsForPromptOutput[]): void {
+  assert.isArray(cfr);
+  for (const item of cfr) {
+    assertContentFilterResultsForPromptItem(item);
+  }
+}
+
+function assertContentFilterCitedDetectionResult(
+  val: ContentFilterCitedDetectionResultOutput,
+): void {
+  assert.isBoolean(val.detected);
+  assert.isBoolean(val.filtered);
+  ifDefined(val.license, assert.isString);
+  ifDefined(val.URL, assert.isString);
+}
+
+function assertContentFilterResultsForPromptItem(cfr: ContentFilterResultsForPromptOutput): void {
+  assert.isNumber(cfr.prompt_index);
+  assertContentFilterResultDetailsForPrompt(cfr.content_filter_results);
+}
+
+function assertContentFilterResultDetailsForPrompt(
+  cfr: ContentFilterResultDetailsForPromptOutput,
+): void {
+  if (cfr.error) {
+    assertContentFilterErrorResults(cfr.error);
+  } else {
+    ifDefined(cfr.hate, assertContentFilterResult);
+    ifDefined(cfr.self_harm, assertContentFilterResult);
+    ifDefined(cfr.sexual, assertContentFilterResult);
+    ifDefined(cfr.violence, assertContentFilterResult);
+    ifDefined(cfr.profanity, assertContentFilterDetectionResult);
+    ifDefined(cfr.jailbreak, assertContentFilterDetectionResult);
+    ifDefined(cfr.custom_blocklists, (arr) =>
+      assertArray(arr, assertContentFilterBlocklistIdResult),
+    );
+  }
+}
+
+function assertContentFilterErrorResults(error: ErrorModel): void {
+  assert.isDefined(error);
+  assert.isDefined(error.code);
+  assert.isDefined(error.message);
+}
+
+function assertContentFilterResult(val: ContentFilterResultOutput): void {
+  assert.isBoolean(val.filtered);
+  assert.isString(val.severity);
+}
+
+function assertContentFilterDetectionResult(val: ContentFilterDetectionResultOutput): void {
+  assert.isBoolean(val.detected);
+  assert.isBoolean(val.filtered);
+}
+
+function assertContentFilterBlocklistIdResult(val: ContentFilterBlocklistIdResultOutput): void {
+  assert.isString(val.id);
+  assert.isBoolean(val.filtered);
+}
+
+function assertChoice(
+  choice: ChatCompletion.Choice | ChatCompletionChunk.Choice,
+  options: ChatCompletionTestOptions,
+): void {
+  const stream = options.stream;
+  if (stream) {
+    assertMessage((choice as ChatCompletionChunk.Choice).delta, options);
+    assert.isFalse("message" in choice);
+  } else {
+    assertMessage((choice as ChatCompletion.Choice).message, options);
+    assert.isFalse("delta" in choice);
+  }
+  assert.isNumber(choice.index);
+  ifDefined(choice.content_filter_results, assertContentFilterResultsForChoice);
+  ifDefined(choice.enhancements, assertAzureChatEnhancements);
+  ifDefined(choice.finish_details, assertChatFinishDetails);
+  ifDefined(choice.logprobs, assertLogProbability);
+  ifDefined(choice.finish_reason, assert.isString);
+}
+
+export async function assertCompletionsStream(
+  stream: AsyncIterable<Omit<OpenAI.Completions.Completion, "usage">>,
```

Comment:
Could we get rid of all these qualified imports? We shouldn't need to import `OpenAI` here.